src/objects.c

```diff
@@ -865,6 +865,17 @@ static UInt LastPV = 0; /* This variable contains one of the values
 			   PrintObj or ViewObj still open (0), or the
 			   innermost such is Print (1) or View (2) */
 
+/* On-demand creation of the PrintObj stack */
+void InitPrintObjStack()
+{
+  if (!TLS(PrintObjThiss)) {
+    TLS(PrintObjThissObj) = NewBag(T_DATOBJ, MAXPRINTDEPTH*sizeof(Obj)+sizeof(Obj));
+    TLS(PrintObjThiss) = ADDR_OBJ(TLS(PrintObjThissObj))+1;
+    TLS(PrintObjIndicesObj) = NewBag(T_DATOBJ, MAXPRINTDEPTH*sizeof(Int)+sizeof(Obj));
+    TLS(PrintObjIndices) = (Int *)(ADDR_OBJ(TLS(PrintObjIndicesObj))+1);
+  }
```

Comment:
Is this thread safe (and, does that matter)?
It seems that there could be a race here, where two threads see that there is no `PrintObjThiss`, and both init it.